src/sql/src/plan/statement/ddl.rs

```diff
@@ -4897,6 +4898,53 @@ pub fn plan_alter_schema_rename(
     }))
 }
 
+pub fn plan_alter_schema_swap<F>(
+    scx: &mut StatementContext,
+    name_a: UnresolvedSchemaName,
+    name_b: UnresolvedSchemaName,
+    gen_temp_suffix: F,
+) -> Result<Plan, PlanError>
+where
+    F: Fn(&dyn Fn(&str) -> bool) -> Result<String, PlanError>,
+{
+    let schema_a = scx.resolve_schema(name_a.clone())?;
+    let schema_b = scx.resolve_schema(name_b.clone())?;
+
+    // Schemas need to be in the same database.
+    let db_spec = if schema_a.database() == schema_b.database() {
+        schema_a.database().clone()
+    } else {
+        sql_bail!("cannot swap schemas that are in different databases");
+    };
+    if matches!(db_spec, ResolvedDatabaseSpecifier::Ambient) {
+        sql_bail!("cannot swap schemas that are in the ambient database");
+    };
+
+    // We cannot swap system schemas.
+    if schema_a.id().is_system() || schema_b.id().is_system() {
+        bail_never_supported!("swaping a system schema".to_string())
+    }
+
+    // Generate a temporary name we can swap schema_a to.
+    //
+    // 'check' returns if the temp schema name would be valid.
+    let check = |temp_suffix: &str| {
+        let temp_name = Ident::new(format!("schema_swap_{temp_suffix}"));
+        scx.resolve_schema_in_database(&db_spec, &temp_name)
+            .is_err()
+    };
+    let temp_suffix = gen_temp_suffix(&check)?;
+    let name_temp = format!("schema_swap_{temp_suffix}");
```

Comment:
You could alternatively just use a hardcoded name like `mz_swap`, I think. Names that start with `mz_` are reserved for the system, and we could just declare that `mz_swap` must always not exist, so that we can use it for this purpose.

Reply:
Sweet, I'll prefix with `mz_`!